pkg/controller/atlasproject/team_reconciler.go

```diff
@@ -0,0 +1,265 @@
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlas"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/statushandler"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func teamReconcile(
+	team *v1.AtlasTeam,
+	kubeClient client.Client,
+	eventRecorder record.EventRecorder,
+	atlasConnection atlas.Connection,
+	atlasDomain string,
+	logger *zap.SugaredLogger,
+) reconcile.Func {
+	return func(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+		log := logger.With("atlasteam", req.NamespacedName)
+
+		result := customresource.PrepareResource(kubeClient, req, team, log)
+		if !result.IsOk() {
+			return result.ReconcileResult(), nil
+		}
+
+		workflowCtx, err := createTeamContextFromParent(team, kubeClient, atlasConnection, atlasDomain, log)
+		if err != nil {
+			workflowCtx.SetConditionFalse(status.ReadyType)
+			return workflow.Terminate(workflow.Internal, err.Error()).ReconcileResult(), nil
+		}
+
+		defer statushandler.Update(workflowCtx, kubeClient, eventRecorder, team)
+
+		log.Infow("-> Starting AtlasTeam reconciliation", "spec", team.Spec)
+
+		teamID, result := ensureTeamState(ctx, workflowCtx, team)
+		if !result.IsOk() {
+			workflowCtx.SetConditionFromResult(status.ReadyType, result)
+			if result.IsWarning() {
+				workflowCtx.Log.Warnf("failed to ensure team state %v: %s", team.Spec, result.GetMessage())
+			}
+
+			return result.ReconcileResult(), nil
+		}
+
+		workflowCtx.EnsureStatusOption(status.AtlasTeamSetID(teamID))
+
+		result = ensureTeamUsesAreInSync(ctx, workflowCtx, teamID, team)
+		if !result.IsOk() {
+			workflowCtx.SetConditionFromResult(status.ReadyType, result)
+			return result.ReconcileResult(), nil
+		}
+
+		if !team.GetDeletionTimestamp().IsZero() {
+			log.Infow("-> Starting AtlasTeam deletion", "spec", team.Spec)
+			_, err := workflowCtx.Client.Projects.Delete(ctx, team.Status.ID)
+			var apiError *mongodbatlas.ErrorResponse
+			if errors.As(err, &apiError) && apiError.ErrorCode == atlas.NotInGroup {
+				log.Infow("team does not exist", "projectID", team.Status.ID)
+				return workflow.Terminate(workflow.TeamDoesNotExist, err.Error()).ReconcileResult(), nil
+			}
+		}
+
+		workflowCtx.SetConditionTrue(status.ReadyType)
+		return workflow.OK().ReconcileResult(), nil
+	}
+}
+
+func createTeamContextFromParent(
+	team *v1.AtlasTeam,
+	kubeClient client.Client,
+	atlasConnection atlas.Connection,
+	atlasDomain string,
+	logger *zap.SugaredLogger,
+) (*workflow.Context, error) {
+	teamCtx := customresource.MarkReconciliationStarted(kubeClient, team, logger)
```

Comment:
I guess it should be first checked if this resource is not marked with `mongodb.com/atlas-reconciliation-policy: skip` label